### PR TITLE
feat: checkbox styles improvement

### LINKED ISSUE #50: Update checkbox component
The checkbox component has a new design, therefore we must update the existing checkbox component file to match the design specifications.

### Design reference
The design specifications for the new design can be found on the following Figma link:
https://www.figma.com/design/XqVOGOoS8tZhGK5ROmqGFv/Open-tasks?node-id=41-73&node-type=section&t=HnHWrabcCxphoBCO-0

### Development
We use shadcn as base for our components implementation, and tailwind for styling. Please stick to both libraries conventions. Make use of class utilities to style the component. For example: set width and height with `w-4 h-4` for the small size and `w-5 h-5` for the medium size.

The existing checkbox component file is located at:
https://github.com/BitcreditProtocol/E-Bill-frontend/blob/revamp/src/components/ui/checkbox.tsx

As a reference on how to implement the variations, please check the following component:
https://github.com/BitcreditProtocol/E-Bill-frontend/blob/revamp/src/components/ui/button.tsx#L10

It's also important to follow the brand guidelines for the colors. The Figma specs already use the naming conventions, and they can be found on the code base at the tailwind configuration file:
https://github.com/BitcreditProtocol/E-Bill-frontend/blob/revamp/tailwind.config.js#L85

### Acceptance criteria:

- [ ] The component styling must match all the states specified in the design specs: default, hover and disabled
- [ ] The component must have two different sizes, as per the design specs: small (sm) and medium (md)
- [ ] There has to be a property to set the checkbox state (checked or unchecked)
- [ ] There has to be a property to set the checkbox size (sm or md)
- [ ] There has to be a property to turn the checkbox into the disabled state

## Important: branch from revamp

### PR DESCRIPTION
**Pull request by nafisreza**
Fixes #50

Screenshots/Video:

https://github.com/user-attachments/assets/2e08432e-178a-4d75-9323-1ed7ccfc521f

Improvements:
- [x] The component styling must match all the states specified in the design specs: default, hover and disabled
- [x] The component must have two different sizes, as per the design specs: small (sm) and medium (md)
- [x] There has to be a property to set the checkbox state (checked or unchecked)
- [x] There has to be a property to set the checkbox size (sm or md)
- [x] There has to be a property to turn the checkbox into the disabled state

### CODE REVIEW

File: src/components/ui/checkbox.tsx [hunk 1]
```diff
@@ -1,25 +1,68 @@
 import * as React from "react"
 import * as CheckboxPrimitive from "@radix-ui/react-checkbox"
-import { Check } from "lucide-react"
+import { Check, Minus } from "lucide-react"
+import { cva, type VariantProps } from "class-variance-authority"
```
Comment (cryptosalomao) [inline src/components/ui/checkbox.tsx:4]: `type VariantProps` is unused.

File: src/components/ui/checkbox.tsx [hunk 2]
```diff
@@ -1,25 +1,68 @@
...
+      // Hover state for unchecked
+      "enabled:hover:bg-elevation-250 enabled:hover:border-divider-75",
+      // Checked/Indeterminate state
+      "data-[state=checked]:bg-brand-50 data-[state=checked]:border-brand-200",
+      "data-[state=indeterminate]:bg-brand-50 data-[state=indeterminate]:border-brand-200",
+      // Hover state for checked/indeterminate
+      "enabled:data-[state=checked]:hover:bg-brand-100 enabled:data-[state=checked]:hover:border-brand-200",
+      "enabled:data-[state=indeterminate]:hover:bg-brand-100 enabled:data-[state=indeterminate]:hover:border-brand-200",
+      // Disabled state
+      "disabled:!bg-[#D1CCC1] disabled:!border-0 disabled:cursor-not-allowed"
```
Comment (cryptosalomao) [inline src/components/ui/checkbox.tsx:51]: Use `bg-elevation-300` here

Comment (cryptosalomao) [inline src/components/ui/checkbox.tsx:51]: These comments are redundant. The data attributes already provide context about what style variation the properties belong to.

**Review by cryptosalomao — CHANGES_REQUESTED** [review]
Hi. Thanks for your contribution! Please review the points I've highlighted

**Comment by nafisreza** [comment]
Fixed. Kindly check now.

**Review by cryptosalomao — APPROVED** [review]
Hi @nafisreza all looking good! Thanks for the contribution!

**Review by mtbitcr — APPROVED** [review]

**Comment by mtbitcr** [comment]
LGTM.
@cryptosalomao you can merge it.
@nafisreza please provide your taproot Bitcoin address for awards. You get 500 e-IOU tokens.

**Comment by nafisreza** [comment]
Thanks a ton @mtbitcr 
I have sent you a DM on Linkedin. Kindly check.

**Comment by mtbitcr** [comment]
@nafisreza please don't use GitHub keywords to close issues automatically. Just link PR to issue

**Pull request merged**